src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2384,21 +2384,37 @@ private bool ConsumeCommentMultiSegment()
                 ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidCharacterAtStartOfComment, marker);
             }
 
+            bool allowEmptyComment = marker == JsonConstants.Slash;
+
             localBuffer = localBuffer.Slice(1);
             if (localBuffer.Length == 0)
             {
-                if (IsLastSpan)
-                {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
-                }
-                if (!GetNextSpan())
+                if (IsLastSpan || !GetNextSpan())
```

Comment:
This condition flow is hard to follow now and has a duplicate check (when `IsLastSpan` == true).

Reply:
it's because GetNextSpan changes the state - if IsLastSpan is already true it should be very cheap to check and compiler is possibly able to detect than no writes were done and optimize this path (if not now then perhaps in the future)

Reply:
I understand the reasoning, but I would prefer to keep the pattern consistent with the rest of the file, especially given the complexity here.

Elsewhere, we have:

```C#
if(IsLastSpan)
{
   // ...
}

if (!GetNextSpan())
{
   if(IsLastSpan)
   {
      // ...
   }
}
```